src/server.c

```diff
@@ -5116,8 +5705,26 @@ void addReplyCommandKeyArgs(client *c, struct redisCommand *cmd) {
     }
 }
 
+void addReplyCommand(client *c, struct redisCommand *cmd);
+
+void addReplyCommandSubCommands(client *c, struct redisCommand *cmd) {
+    if (!cmd->subcommands_dict) {
+        addReplySetLen(c, 0);
+        return;
+    }
+
+    addReplySetLen(c, dictSize(cmd->subcommands_dict));
```

Comment:
i think this is an array, not a set.
each entry is a map

Reply:
why array? elements are unique and there's no order

Reply:
it's right that each element is unique (it's a big dictionary with lots of fields, and there are no duplicate records), but i think a set is a set of simple objects like strings.
we could have made it a map (if we moved the name outside the dict).
i.e. a map where each entry has a command name (the map's field), and a value that's an array (or map) of metadata.
but the way it is now, where each entry is a map, doesn't make sense to me as a set.
if you disagree, let's seek a 3rd opinion.

Reply:
i don't mind... @yossigo WDYT? the output of COMMANDS INFO should be a set, list, or hash (key is command name, value is its data)

Reply:
ok, changed it to hash (map) - LMK if you wish me to revert